src/mtp_cmd_handler.c

```diff
@@ -1289,6 +1315,20 @@ static void __process_commands(mtp_handler_t *hdlr, cmd_blk_t *cmd)
                 __begin_end_edit_object(hdlr);
                 break;
 
+	case PTP_OC_ANDROID_SENDPARTIALOBJECT:
+		if (g_device->phase == DEVICE_PHASE_IDLE) {
+			_eh_send_event_req_to_eh_thread(EVENT_START_DATAOUT, 0, 0, NULL);
+			_device_set_phase(DEVICE_PHASE_DATAOUT);
+			return;
+		}
+		_util_file_copy(g_copy_src_file, g_copy_dst_file, &error);
```

Comment:
This deserves explanation - why receiving a partial object by an mtp device equals copying a file.

Reply:
> This deserves explanation - why receiving a partial object by an mtp device equals copying a file.

This is a workaround, will fix it in next patch. Actually was trying to see if with support of below MTP events, whether rename and duplication works.

SendPartialObject : allows writing a subset of a file, or appending to the end of a file
BeginEditObject :  must be called before using SendPartialObject and TruncateObject
EndEditObject : commits changes to a file after it has been edited with SendPartialObject or TruncateObject

Reply:
The series seems incomplete with the next patch, then.

Reply:
True, currently it is a workaround. 
Will fix this in next commit asap. In this patch was trying to focus on all missing MTP events